pkg/api.go

```diff
@@ -1,44 +1,55 @@
 package pkg
 
 import (
+	"fmt"
 	"net/http"
 
 	"github.com/flanksource/commons/logger"
 	"k8s.io/apimachinery/pkg/util/json"
 )
 
-type Input struct {
-	// list of environments where config can be applied
-	Environments []string
-	// list of services of items being templated.
-	Applications []string
-	// Resources that are generated by Kustomize
-	Resources []Resource
-	// The hierarchy in which configs are imported
-	Config Config
-}
-
 type APIServer struct {
-	Repos     []string
-	Branches  []string
-	Hierarchy Config
+	Repos      []string
+	Branches   []string
+	ConfigFile string
 }
 
-func (input *APIServer) Handler(w http.ResponseWriter, req *http.Request) {
-	showObjects := req.URL.Query().Get("objects")
-	data := ReadConfiguration(input.Repos, input.Branches, input.Hierarchy)
+func GetConfigData(input *APIServer, appName string, showObjects string) ([]map[string]KustomizeResources, error) {
+	hierarchy, err := GetHierarchy(input.ConfigFile, appName)
+	if err != nil {
+		return nil, err
+	}
+	data := ReadConfiguration(input.Repos, input.Branches, hierarchy)
+
 	if showObjects == "" {
 		data = removeObjectsFromList(data)
 	}
-	output, err := json.Marshal(data)
-	if err != nil {
-		logger.Fatalf("error marshalling the data")
-		return
-	}
-	_, err = w.Write(output)
-	if err != nil {
-		logger.Fatalf("Error writing the body")
-		return
+
+	return data, nil
+}
+
+func (input *APIServer) GetConfigHandler() http.HandlerFunc {
+	return func(w http.ResponseWriter, req *http.Request) {
+		w.Header().Set("Content-Type", "application/json")
```

Comment:
```suggestion
		resp.Header().Set("Content-Type", "application/json")
```

Single letter var names are for loop indexes only

Reply:
Changed